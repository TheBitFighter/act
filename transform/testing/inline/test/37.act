function flip_msb(int<4> x) : int<8>
{ 

  chp {
    x := {~x{3}, x{2..0}};
    self := x
  }
}

defproc test()
{
  int<8> x;
  int<8> y;
  chp{
    x := 0;
    y := flip_msb(x{3..0}) ; log("%b", y) 

  }
}

test t;
