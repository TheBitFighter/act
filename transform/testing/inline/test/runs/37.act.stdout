defproc test ();
function flip_msb (int<4> x) : int<8>;

defproc test ()
{

/* instances */
int<8> y;
int<8> x;

/* connections */
chp {
x:=0x0;y:= {~int(x{0x3..0x0}>>0x3&1,1),int(x{0x3..0x0}>>0x0&7,3)};log("%b",y)
}
}

function flip_msb (int<4> x) : int<8>
{

/* instances */

/* connections */
chp {
x:= {~x{0x3},x{0x2..0x0}};self:=x
}
}


/* instances */
test t;

/* connections */
